gridcapa-core-cc-post-processing-app/src/test/java/com/farao_community/farao/core_cc_post_processing/app/services/RaoIXmlResponseGeneratorTest.java

```diff
@@ -0,0 +1,156 @@
+    private CoreCCMetadata coreCCMetadataRunningTask = new CoreCCMetadata("raoRequest.json", "2023-08-04T11:26:00Z", "2023-08-04T11:26:00Z", "2023-08-04T11:27:00Z", "2023-08-04T11:29:00Z", "2023-08-04T11:25:00Z/2023-08-04T12:25:00Z", "correlationId", "SUCCESS", "0", "This is an error.", 1);
+    private CoreCCMetadata coreCCMetadataSuccessTask = new CoreCCMetadata("raoRequest.json", "2023-08-04T11:26:00Z", "2023-08-04T11:26:00Z", "2023-08-04T11:27:00Z", "2023-08-04T11:29:00Z", "2023-08-04T11:25:00Z/2023-08-04T12:25:00Z", "correlationId", "SUCCESS", "0", "This is an error.", 1);
+    private UUID uuidErrorTask = UUID.fromString("22711acb-ee59-47ed-b877-3c3688efe820");
+    private UUID uuidRunningTask = UUID.fromString("259ebbe3-4639-4fa6-9687-cdb38a2f36cc");
+    private UUID uuidSuccessTask = UUID.fromString("6df6092d-145b-47be-b412-54fc45a90a04");
+    private Map<UUID, CoreCCMetadata> metadataMap = new HashMap<UUID, CoreCCMetadata>();
+    private ProcessFileDto processFileDto = new ProcessFileDto("/path/network.uct", "CGM_OUT", ProcessFileStatus.VALIDATED, "network.uct", OffsetDateTime.of(2023, 8, 4, 16, 39, 00, 0, ZoneId.of("Europe/Brussels").getRules().getOffset(LocalDateTime.now())));
+
+    @BeforeEach
+    void setUp() {
+        minioAdapter = Mockito.mock(MinioAdapter.class);
+        raoResponseIsUploadedToMinio = false;
+    }
+
+    @Test
+    void generateCgmXmlHeaderFile() throws IOException {
+        initTasksForCgmXmlHeader();
+        RaoIXmlResponseGenerator raoIXmlResponseGenerator = new RaoIXmlResponseGenerator(minioAdapter);
+        File generatedXmlHeaderFile = new File(cgmsArchiveTempPath, "CGM_XML_Header.xml");
+        // First pass with not existing directory
+        compareOriginalAndGeneratedXmlHeaders(raoIXmlResponseGenerator, generatedXmlHeaderFile);
+        // Second pass with already existing directory
+        compareOriginalAndGeneratedXmlHeaders(raoIXmlResponseGenerator, generatedXmlHeaderFile);
+        // Delete the temporary directory
+        FileUtils.deleteDirectory(new File(generatedXmlHeaderFile.getParent()));
+        assertFalse(Files.exists(generatedXmlHeaderFile.getParentFile().toPath()));
+    }
+
+    private void initTasksForCgmXmlHeader() {
+        taskDtoStart = Mockito.mock(TaskDto.class);
+        Mockito.doReturn(startInstant).when(taskDtoStart).getTimestamp();
+        Mockito.doReturn(TaskStatus.SUCCESS).when(taskDtoStart).getStatus();
+        taskDtoEnd = Mockito.mock(TaskDto.class);
+        Mockito.doReturn(endInstant).when(taskDtoEnd).getTimestamp();
+        Mockito.doReturn(TaskStatus.SUCCESS).when(taskDtoEnd).getStatus();
+        taskDtos = Set.of(taskDtoStart, taskDtoEnd);
+    }
+
+    private void changeDateInGeneratedCggmXmlHeader(File file) throws IOException {
```

Comment:
I added 4 UTs using the sources' functions for the payload and the header of both the RaoResponse and the Cgm.

Yet, I chose to keep the `generateCgmXmlHeaderFile` test relying on the `changeDateInGeneratedCggmXmlHeader` custom method because this file has two different behaviours depending on whether the target directory exists or not.